crates/dogoap/src/planner.rs

```diff
@@ -47,72 +50,37 @@ fn successors<'a>(
     let state = node.state();
     actions.iter().filter_map(move |action| {
-        if check_preconditions(state, action) && !action.effects.is_empty() {
-            let new_state = state.clone();
-            let first_effect = &action.effects[0];
-
-            let mut new_data = new_state.data.clone();
-            for mutator in &first_effect.mutators {
-                apply_mutator(&mut new_data, mutator);
-            }
+        if !check_preconditions(state, action) || action.effects.is_empty() {
+            return None;
+        }
+        let new_state = state.clone();
+        let first_effect = &action.effects[0];
 
-            let new_effect = Effect {
-                action: first_effect.action.clone(),
-                mutators: first_effect.mutators.clone(),
-                cost: first_effect.cost,
-                state: LocalState { data: new_data },
-            };
-            Some((Node::Effect(new_effect), first_effect.cost))
-        } else {
-            None
+        let mut new_data = new_state.data.clone();
+        for mutator in &first_effect.mutators {
+            apply_mutator(&mut new_data, mutator);
         }
+
+        let new_effect = Effect {
+            action: first_effect.action.clone(),
+            mutators: first_effect.mutators.clone(),
+            cost: first_effect.cost,
+            state: LocalState { data: new_data },
+        };
+        Some((Node::Effect(new_effect), first_effect.cost))
     })
 }
 
 fn is_goal(node: &Node, goal: &Goal) -> bool {
     goal.requirements.iter().all(|(key, value)| {
-        if let Some(state_val) = node.state().data.get(key) {
-            compare_values(value, state_val)
-        } else {
-            panic!("Couldn't find key {:#?} in LocalState", key);
-        }
-    })
-}
-
-/// Use [`make_plan`] instead
-pub fn make_plan_with_strategy(
-    strategy: PlanningStrategy,
-    start: &LocalState,
-    actions: &[Action],
-    goal: &Goal,
-) -> Option<(Vec<Node>, usize)> {
-    match strategy {
-        PlanningStrategy::StartToGoal => {
-            let start_node = Node::State(start.clone());
-            pathfinding::directed::astar::astar(
-                &start_node,
-                |node| successors(node, actions).collect::<Vec<_>>().into_iter(),
-                |node| heuristic(node, goal),
-                |node| is_goal(node, goal),
+        let state_val = node.state().data.get(key).unwrap_or_else(|| {
+            panic!(
+                "Couldn't find key {key:#?} in LocalState {:#?}",
+                node.state().data
             )
-        }
-        PlanningStrategy::GoalToStart => {
-            panic!("PlanningStrategy::GoalToStart hasn't been implemented yet!");
-        }
-    }
-}
-
-/// Currently, only [`PlanningStrategy::StartToGoal`] is supported, which tries to find the chain of
-/// [`Effect`]s that lead to our [`Goal`] state
-#[derive(Default)]
-pub enum PlanningStrategy {
```

Comment:
Lets keep this around. It's true that `GoalToStart` hasn't been implemented, but removing the whole thing doesn't really solve that particular problem ;)